lib/ddtrace/contrib/rack/request_queue.rb

```diff
@@ -12,14 +12,14 @@ def get_request_start(env, now = Time.now.utc)
           header = env[REQUEST_START] || env[QUEUE_START]
           return unless header
 
-          # nginx header is in the format "t=1512379167.574"
-          # TODO: this should be generic enough to work with any
-          # frontend web server or load balancer
-          time_string = header.split('t=')[1]
+          # nginx header is milliseconds in the format "t=1512379167.574"
+          # apache header is microseconds in the format "t=1570633834463123"
+          # heorku header is milliseconds in the format "1570634024294"
+          time_string = header.to_s.gsub(/t=|\./, '')
```

Comment:
I was curious if this was the best way to remove all non-numeric characters from a string, so I [ran a benchmark](https://gist.github.com/marcotc/09db9af991b0ad3c435573836f86221e).

`String#delete` seems to be the fastest way, also using the least amount of memory, so I suggest this change:

```suggestion
          time_string = header.to_s.delete('^0-9')
```